Update CI/CD doc with high level overview and additional GitHub actions info (#273)
astronomer/docs: enterprise/v0.23/04_deploy/06_ci-cd.md

In this example:

> - `BASEDOMAIN` = `gcp0001.us-east4.astronomer.io`

Reviewer: same

## Step 3: Configure Your CI/CD Pipeline

Depending on your CI/CD tool, configuration will be slightly different. This section will focus on outlining what needs to be accomplished, not the specifics of how.

At its core, your CI/CD pipeline will first authenticate to Astronomer's private registry and then build, tag and push your Docker Image to that registry.

## Example Implementation

The following setup is an example implementation of CI/CD using GitHub Actions. These steps cover both the implementation and the workflow necessary to create a fully functional CI/CD pipeline.

1. Create a GitHub repository for an Astronomer project. Ensure your repo has a development branch and a main branch. In this example, the branches are named `dev` and `main`.
2. [Create a Service Account](/docs/enterprise/v0.23/deploy/ci-cd#step-1-create-a-service-account) for your Astronomer Workspace.
3. Follow instructions in [GitHub documentation](https://docs.github.com/en/actions/reference/encrypted-secrets) to add your Astronomer Service Account as a secret to your repository. In the example below, that secret is named `SERVICE_ACCOUNT_KEY`.

> 4. Go to the Actions tab of your GitHub repo and create a new action with a `main.yml` file. To achieve the recommended workflow described in [Overview](/docs/cloud/deploy/ci-cd#overview), use the following action:

Reviewer: ```suggestion
4. Go to the Actions tab of your GitHub repo and create a new action with a `main.yml` file. To achieve the recommended workflow described in [Overview](/docs/enterprise/v0.23/deploy/ci-cd#overview), use the following action:
```

    ```yaml
    name: CI
    on:
      push:
        branch: [dev, master]
    jobs:
      dev-push:
        runs-on: ubuntu-latest
        steps:
        - uses: actions/checkout@v1
        - name: Push to registry
          uses: elgohr/Publish-Docker-Github-Action@2.6
          if: github.ref == 'refs/heads/dev'
          with:
              name: <dev-release-name>/airflow:ci-${{ github.sha }}
              username: _
              password: ${{ secrets.SERVICE_ACCOUNT_KEY }}
              registry: registry.${BASE_DOMAIN}
      prod-push:
        runs-on: ubuntu-latest
        steps:
        - uses: actions/checkout@v1
        - name: Push to registry
          uses: elgohr/Publish-Docker-Github-Action@2.6
          if: github.ref == 'refs/heads/master'
          with:

>               name: <prd-release-name>/airflow:ci-${{ github.sha }}

Reviewer: ```suggestion
              name: <prod-release-name>/airflow:ci-${{ github.sha }}
```

              username: _
              password: ${{ secrets.SERVICE_ACCOUNT_KEY }}
              registry: registry.${BASE_DOMAIN}
    ```

    Ensure the branches match the names of the branches in your repository, and replace `<dev-release-name>` and `<prod-release-name>` with the respective release names of your development and production Airflow Deployments on Astronomer.

5. Test the GitHub Action by making a change on your `dev` branch and committing that change. This should update your development Airflow Deployment on Astronomer, which you can confirm in the Astronomer UI. If that update was successful, try then merging `dev` into `master` to update your production Airflow Deployment. If both updates were successful, you now have a functioning, scalable CI/CD pipeline that can automatically deploy code to multiple Airflow Deployments.

> > **Note:** The prod-push action as defined here will run on any push to the `master` branch, including a pull request and merge from the `dev` branch as we recommend.
> >
> >To further restrict this to run only on a pull request, you can limit whether your users can push directly to the `master` branch within your repository or your CI tool, or you could modify the action to make it more limited.

Reviewer: ```suggestion
> **Note:** The prod-push action as defined here will run on any push to the `main` branch, including a pull request and merge from the `dev` branch as we recommend.
>
>To further restrict this to run only on a pull request, you can limit whether your users can push directly to the `main` branch within your repository or your CI tool, or you could modify the action to make it more limited.
```
